ci/cscs-ci.yml

```diff
@@ -2,37 +2,72 @@ include:
   - remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
 
 stages:
-  - build
+  - deps
+  - cp2k
   - test
   - benchmark
 
 variables:
-  DOCKERHUB_REPOSITORY: cp2k_eiger
-  DOCKERHUB_TAG: toolchain_psmp
-  DOCKERHUB_IMAGE_NAME: docker.io/${DOCKERHUB_USERNAME}/${DOCKERHUB_REPOSITORY}:${DOCKERHUB_TAG}
-  PERSIST_IMAGE_NAME: ${CSCS_REGISTRY_PATH}/cp2k:${CI_COMMIT_SHORT_SHA}
+  BASE_IMAGE: "ubuntu:24.04"
+  BUILD_TYPE: "toolchain"
+  DOCKERHUB_REPOSITORY: "cp2k_eiger"
+  DOCKERHUB_TAG: "toolchain_psmp"
+  DOCKERHUB_BASE_IMAGE_NAME: "docker.io/${DOCKERHUB_USERNAME}/${DOCKERHUB_REPOSITORY}:${DOCKERHUB_TAG}_base_image"
+  DOCKERHUB_IMAGE_NAME: "docker.io/${DOCKERHUB_USERNAME}/${DOCKERHUB_REPOSITORY}:${DOCKERHUB_TAG}"
+  PERSIST_IMAGE_NAME: "${CSCS_REGISTRY_PATH}/cp2k:${CI_COMMIT_SHORT_SHA}"
+
+build deps eiger:
+  extends: .container-builder-cscs-zen2
+  stage: deps
+  timeout: 1h
+# rules:
+#   - changes:
+#     - tools/toolchain/scripts/install_cp2k_toolchain.sh
+#     - tools/toolchain/scripts/**/*
+  before_script:
+    - echo ${DOCKERHUB_TOKEN} | podman login docker.io --username ${DOCKERHUB_USERNAME} --password-stdin
+  after_script:
+    - podman images
+    - podman push ${PERSIST_IMAGE_NAME} ${DOCKERHUB_BASE_IMAGE_NAME}
```

Comment:
FYI, instead of pushing the images manually, it is possible to use the builtin feature of pushing to multiple registries. See [third party docker registries](https://confluence.cscs.ch/spaces/KB/pages/868812119/Third+party+docker+registries).

Reply:
That builtin option does not simplify the current pipeline workflow, because we have to push and retain two images concurrently.

Reply:
> FYI, instead of pushing the images manually, it is possible to use the builtin feature of pushing to multiple registries. See [third party docker registries](https://confluence.cscs.ch/spaces/KB/pages/868812119/Third+party+docker+registries)

The login with `SECONDARY_IMAGE_USERNAME` and `SECONDARY_IMAGE_PASSWORD` [does not work](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/3546923033037693/3295760759903019/-/jobs/9309594551).